core/trino-main/src/main/java/io/trino/operator/scalar/JoniRegexpFunctions.java

```diff
@@ -408,4 +409,20 @@ public static long regexpCount(@SqlType("varchar(x)") Slice source, @SqlType(Jon
 
         return count;
     }
+
+    public static int getSearchingOffset(Matcher matcher, int at, int range)
+    {
+        try {
+            return matcher.searchInterruptible(at, range, Option.DEFAULT);
+        }
+        catch (InterruptedException interruptedException) {
+            // The JONI library is compliant with the InterruptedException contract. They reset the interrupted flag before throwing an exception.
+            // Since the InterruptedException is being caught the interrupt flag must either be recovered or the thread must be terminated.
+            // Since we are simply throwing a different exception, the interrupt flag must be recovered to propagate the interrupted status to the upper level code.
```

Comment:
This sounds redundant. 
We have this pattern (`catch (InterruptedException) + Thread.currentThread().interrupt() + throw`) in many places, and it usually (always?) goes without a comment. It's an idiom.

Reply:
Okay, I'll remove all of those.

Reply:
Removed in #13229